polkadot/consensus/src/service.rs

```diff
@@ -211,21 +211,23 @@ pub struct Service {
 
 impl Service {
 	/// Create and start a new instance.
-	pub fn new<C>(
+	pub fn new<A, C>(
```

Comment:
i'm not convinced that this service needs to support light client in any form since it's only for validators (who are definitely full nodes)

Reply:
there should be something of a distinction between client backend types where we can bound things by `Backend: LocalBackend` to filter out light nodes.

Reply:
Added [here](https://github.com/paritytech/polkadot/pull/150/commits/a1137fcf0a91758a07ca6183cfead14f905ce8bf). Had to add another pair of traits (`LocalPolkadotApi` + `RemotePolkadotApi`), since consensus service depends on `PolkadotApi` trait (and I didn't wanted to make `PolkadotApi` generic).